### ISSUE TITLE
[Bug] iOS - Offline - Stuck comments occur after queued comments send on return from offline

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#10615]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Run NewDot App
2. Open any DM chat and turn network off
3. Send few messages and turn network on
4. Send and receive some messages

## Expected Result:
The user should be able to send messages while offline, they should be delivered correctly once network restored

## Actual Result:
Messages that were sent while offline are stuck at the very bottom. New messages are displayed in the wrong position or not displayed at all

## Workaround:
Unknown

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:** 1.1.92.0

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**Email or phone of affected tester (no customers):** any

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/93399543/187032408-df2220ff-0494-481c-b791-d71e55f67b26.mp4
https://user-images.githubusercontent.com/93399543/187032413-eaa567ec-78b8-4855-840f-6df23fcf6a85.mp4

**Expensify/Expensify Issue URL:** 

**Issue reported by:** Applause - Internal Team

**Slack conversation:** 

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by arosiclair** [comment]
cc @roryabraham @marcaaron @danieldoglas 

Would our new `NumberUtils.rand64()` IDs usage affect the ordering of comments like this?

**Comment by danieldoglas** [comment]
@arosiclair the ordering could be related since [this](https://github.com/Expensify/Auth/pull/7002) deploy was reverted and I don't think we've deployed it back again?

But I've also tested here and it looks like we're trying to do the write requests even when we're offline, which is strange, so It might be something else too!

**Comment by marcaaron** [comment]
>Would our new NumberUtils.rand64() IDs usage affect the ordering of comments like this?

No, because we are still ordering by `sequenceNumber` or `clientID` IIRC.

Are we sure these aren't just "stuck comments" i.e. comments that were created in the server but a Pusher update never received? We've seen that happen recently.

**Comment by marcaaron** [comment]
Looking at the video and I am 99% sure that's what is happening... see how the comments are not just out of order but duplicated as well...

<img width="392" alt="Screen Shot 2022-08-29 at 11 10 19 AM" src="https://user-images.githubusercontent.com/32969087/187166863-21a408b8-b15a-4dcb-b435-2c74d9a7f2cf.png">

**Comment by marcaaron** [comment]
There is no issue with ordering seems like the ordering works just stuck comments.

**Comment by arosiclair** [comment]
Hmm thanks for the insights. It seems like these optimistic comments might always have the highest `sequenceNumber`. I'll attempt to repro this and see if I can dig in further

**Comment by danieldoglas** [comment]
@AndrewGable maybe this is related? https://github.com/Expensify/App/issues/10524

**Comment by JmillsExpensify** [comment]
@arosiclair Let's hold on this one, as it's related to a design doc on `sequenceNumbers` that @roryabraham and @tylerkaraszewski are working on.

**Comment by arosiclair** [comment]
Still on hold

**Comment by arosiclair** [comment]
Still on hold.

@roryabraham you said you wouldn't mind picking up an issue like this since you're working on killing sequence numbers?

**Comment by mallenexpensify** [comment]
For reference, from #expensify-open-source - [link](https://expensify.slack.com/archives/C01GTK53T8Q/p1663880640472169?thread_ts=1663714013.290919&cid=C01GTK53T8Q)

Many duplicate messages showing on the sender side, but not receiver.  Tough to reproduce

**Comment by mallenexpensify** [comment]
@arosiclair , here's the tracking issue for deprecating sequence numbers https://github.com/Expensify/Expensify/issues/227234

**Comment by mallenexpensify** [comment]
Understandable @mandyalvey , the _good_ part is it's only on your end, the other person doesn't think you realllly want to thank them a lot. 

Updated title to include [HOLD Sequence #] so we can keep tabs on

**Comment by arosiclair** [comment]
I'll leave this in the pool to be picked up when sequence numbers are killed

**Comment by zanyrenney** [comment]
<img width="879" alt="2022-10-20_17-26-59" src="https://user-images.githubusercontent.com/56830058/197005388-bb3f92b1-0b2a-42ca-a373-bd5627cd6ef7.png">
Still happening.

**Comment by stitesExpensify** [comment]
I think we should keep this on hold for now since we are about to release optimistic reports as well

**Comment by JmillsExpensify** [comment]
Agreed, I added this issue to the sequence numbers tracking issue.

**Comment by stitesExpensify** [comment]
Still on hold. Making this weekly so it doesn't ping every day

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by stitesExpensify** [comment]
not reproducible and on hold

**Comment by JmillsExpensify** [comment]
If it's not reproducible, let's add the label and close, yeah?

**Comment by stitesExpensify** [comment]
I usually wait until it's not reproducible twice in a row. Is that not the standard process? (genuinely asking)

**Comment by JmillsExpensify** [comment]
Oh yeah, same here. I usually wait twice, I just saw Applause's comment and then your comment, so I thought that was enough to close this out.

**Comment by JmillsExpensify** [comment]
Also not sure why there isn't a BZ on this issue, so I assigned myself and we can align on next steps Monday.

**Comment by stitesExpensify** [comment]
Ah I can totally see the confusion there. I was just replying to the overdue label. I didn't specifically test it and find it unreproducible. That said, I think that we _should_ actually just close this because it will be fixed by our sequence number fixes

**Comment by JmillsExpensify** [comment]
Oh hahaha. Gotcha, yeah I experience this bug still a decent amount. I agree that it's going to get fixed by sequence numbers, but I'd rather keep it open so that it's clear to the community that this is a long-known existing bug and any future reports are duplicates. I'll take the work of commenting weekly.

**Comment by stitesExpensify** [comment]
Sounds like a plan!

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by JmillsExpensify** [comment]
This is still a sporadic issue. Keeping the issue open and on hold.

**Comment by JmillsExpensify** [comment]
Still holding.

**Comment by JmillsExpensify** [comment]
Removed iPad from the title since that's not officially supported, though we're still waiting on the Sequence Numbers.

**Comment by JmillsExpensify** [comment]
Still waiting on sequence numbers.

**Comment by JmillsExpensify** [comment]
Still held on sequence numbers.

**Comment by stitesExpensify** [comment]
Not overdue, I'm unsure why Jason's comment didn't remove the label...

**Comment by JmillsExpensify** [comment]
Woah yeah that's pretty weird.

**Comment by JmillsExpensify** [comment]
Still on hold for sequence numbers but hopefully that'll we'll be unblocked soon and can circle back on testing.

**Comment by JmillsExpensify** [comment]
Held on sequence numbers.

**Comment by JmillsExpensify** [comment]
On hold Melvin. "Hold" your horses.

**Comment by JmillsExpensify** [comment]
Still on hold for SN.

**Comment by JmillsExpensify** [comment]
Removing hold and re-testing.

**Comment by JmillsExpensify** [comment]
Unable to reproduce now. Closing and adding the reproduction label.
